Upstatement/ups-public-library
# Add Table of Contents to entries

# What this does

- Adds an interactive table of contents to entries, with smooth scrolling between anchors and an active anchor indicator. Markup is automatically generated using the plugin [eleventy-plugin-nesting-toc](https://www.npmjs.com/package/eleventy-plugin-nesting-toc)
- reconfigures markdown-it to remove anchor tags from headings
- widens the entry navbar
- adds last updated date & topic to entries
- updates the layout of entry footers
- adds discrete `topic` attributes to each entry, rather than in the `tags` list
- converts the content max width variable SCSS to a CSS variable, makes it responsive
- increases the breakpoint at which the largest type scale is used

# What's included
- installs [eleventy-plugin-nesting-toc](https://www.npmjs.com/package/eleventy-plugin-nesting-toc)
- new class, `InteractiveTOC`

# How to Test
0. `npm install`
1. Go to the [Intro to Terminal](http://localhost:8080/entries/intro-to-terminal/) entry
2. scroll through the entry. the column of bullets on the left should update as you pass headings
3. hover over the column of bullets and click on a heading. you should be smoothly scrolled to that heading
4. shrink your viewport so that the table of contents scrolls. it should become scrollable when you hover over it.
5. scroll the table of contents so that the anchor for next heading you'll see is out of view. the table should autoscroll the anchor into view (this only seems to work when using a mouse's scroll wheel, and not on trackpads)
